### DISCUSSION TITLE
What you doing when Target disappear when Dropdown is opened?

### DISCUSSION THREAD

**Discussion by 7iomka** [mantinedev/mantine#4694] [Q&A]
For many components with Dropdown I have an issue: when Target disappear while Dropdown is opened - dropdown position is recalculated by floating ui, and because target is hidden it use fallback position to start of nearest relative element, for cases withinPortal - it is the start of window.
Expected behavior for this common case: Dropdown need to be hidden until target is hidden.

Guys, how you solve this issue?
There have been similar questions already but I still don't see a solution being found and working out of the box.

**Comment by rtivital** [comment]
It's really hard to do anything in this case except using a controlled Popover state.

**Comment by 7iomka** [comment]
@rtivital In my case all components are controlled. 
But conditions of visibility of Target are very diverse!
In my case in some scenarios Target parent is hidden on specific breakpoint by className (tailwind/css module).
So we need each time to wrap Popover/Menu/Another component that uses dropdown within portal and somehow in useEffect to calculate visibility of Target.
This is a highly undesirable action for me as a ui library user expecting this to be handled by the library itself once and for all.
If it's not possible to include this in the library I'd like help from the community - is there any way to fix this behaviour globally?

**Reply by rtivital** [reply]
You are welcome to submit a PR with a fix
